Swiftfin/Views/PagingLibraryView/PagingLibraryView.swift

```diff
@@ -251,27 +258,59 @@ struct PagingLibraryView<Element: Poster>: View {
         .proxy(collectionVGridProxy)
     }
 
-    // MARK: body
-
-    // TODO: becoming too large for typechecker during development, should break up somehow
+    @ViewBuilder
+    private func contentLetterBarView(content: some View) -> some View {
+        if letterPickerEnabled && viewModel.filterViewModel != nil {
+            switch letterPickerOrientation {
+            case .trailing:
+                HStack(spacing: 0) {
+                    content
+                        .frame(maxWidth: .infinity)
+
+                    LetterPickerBar(viewModel: viewModel.filterViewModel!)
+                        .padding(.top, safeArea.top)
+                        .padding(.bottom, safeArea.bottom)
+                }
+            case .leading:
+                HStack(spacing: 0) {
+                    LetterPickerBar(viewModel: viewModel.filterViewModel!)
+                        .padding(.top, safeArea.top)
+                        .padding(.bottom, safeArea.bottom)
+
+                    content
+                        .frame(maxWidth: .infinity)
+                }
+            }
+        } else {
+            content
+        }
+    }
 
-    var body: some View {
+    @ViewBuilder
+    private func bodyContentView() -> some View {
         ZStack {
             switch viewModel.state {
             case .content:
                 if viewModel.elements.isEmpty {
-                    L10n.noResults.text
+                    contentLetterBarView(content: L10n.noResults.text)
                 } else {
-                    contentView
+                    contentLetterBarView(content: contentView)
                 }
             case let .error(error):
                 errorView(with: error)
             case .initial, .refreshing:
-                DelayedProgressView()
+                contentLetterBarView(content: DelayedProgressView())
             }
         }
         .animation(.linear(duration: 0.1), value: viewModel.state)
         .ignoresSafeArea()
+        .onSizeChanged { _, safeArea in
+            if LetterPickerBar.isScrolling {
```

Comment:
Why the `if` and difference check?

Reply:
At least when I was using the Simulator, when I was using English, the onSizeChanged would recursively call until Swiftfin crashed. The safeArea was changing by like 0.0001 and it would re-call this directly after setting it. I tried with a couple different devices in the Simulator and they all had that same issue for English but not when it was a Scroll when I was using Korean as the language. 

This could totally just be something on my end but this resolved it. I have no idea why the safe area was changing every time it got set with the static HStack but not with the Scroll?? I can take another look / record what I am seeing tomorrow if it's still happening.

Reply:
Interesting, sorry I didn't test with that but glad you were able to figure it out!